PyAssembler/Tutorial1.py

```
@@ -0,0 +1,170 @@
+      elif addr_mode[1].isnumeric():
+         translated_word2 = format(int(addr_mode[1]), '016b')                          # update numeric value bits
+      else:
+         translated_word2 = addr_mode[1]                                               # update placeholder bits
+   elif(line_type == 3):                              # mov cmp add sub lea xor or rol
+      bin_reg1 = switch_register(words[1][:-1])                                        # remove ','
+      if not bin_reg1: invokeError("Invalid first register of 3-type line")
+      else: translated_word1 = translated_word1[:4] + bin_reg1 + translated_word1[7:]  # update first reg bits
+      bin_reg2 = switch_register(addr_mode[1])
+      if bin_reg2:
+         current_address+=2
+         return translated_word1[:9] + bin_reg2 + translated_word1[12:]                # update second reg bits
+      elif addr_mode[1].isnumeric():
+         translated_word2 = format(int(addr_mode[1]), '016b')                          # update numeric value bits
+      else:
+         translated_word2 = addr_mode[1]                                               # update placeholder bits
+   else:
+      invokeError("It is not a valid assembly line (maybe comment or sections).")
+   current_address+=4
+   return '' + translated_word1 + '\n' + translated_word2
+
+
+def fix_line(damaged_line):
+   global sym_tbl
+   try:
+      int(damaged_line, 2)
+      return damaged_line
+   except Exception:
+      for label_tuple in sym_tbl:
+         damaged_line = damaged_line.replace(label_tuple, str(sym_tbl[label_tuple]))
+      damaged_line = eval(damaged_line)
+      return format(int(damaged_line) & 0xffff, '016b')
+
+def section_handler(words):
+   global current_address
+   if(words[0] == '.string'):
+      return convert_str_to_binary(words[1][1:-1])
+   elif(words[0] == '.data'):
+      current_address+=2
+      return format(int(words[1]), '016b')
+   else: invokeError("An invalid opcode was detected.")

```

You could raise an Exception instead of using invokeError (`raise Exception("An invalid opcode was detected.")`)